rclpy/rclpy/executors.py

```diff
@@ -24,6 +24,12 @@
 from rclpy.utilities import ok
 from rclpy.utilities import timeout_sec_to_nsec
 
+# TODO(wjwwood): make _rclpy_wait(...) thread-safe
+# Executor.spin_once() end's up calling _rclpy_wait(...), which right now is
+# not thread-safe, no matter if differnet wait sets are used or not.
+g_wait_set_spinning_lock = Lock()
+g_wait_set_spinning = False
```

Comment:
Note that `_rcply_wait` was not thread-safe before, but my change also does not make it thread-safe, so I thought it would be good to prevent the user from trying to use it that way.

Reply:
I added https://github.com/ros2/rclpy/pull/191/commits/c1870a216c627b4c7d4436fa5a6e52e85a9ca942 to link to an example but if there are bigger issues that should be mentioned (so mentioning that specific one is misleading) we can revert it

Reply:
Do you intend to update the issue you linked to with an example? For now I'd say just `See: <link to issue>` would be fine.

Reply:
I am not sure if I have understood you, but if by example you mean rclpy code example, I was gonna link to a specific commit once this PR was merged